src/test/fuzz/tx_pool.cpp

```diff
@@ -116,14 +116,15 @@ void MockTime(FuzzedDataProvider& fuzzed_data_provider, const CChainState& chain
     SetMockTime(time);
 }
 
-CTxMemPool MakeMempool(const NodeContext& node)
+CTxMemPool MakeMempool(FuzzedDataProvider& fuzzed_data_provider, const NodeContext& node)
 {
     // Take the default options for tests...
     CTxMemPool::Options mempool_opts{MemPoolOptionsForTest(node)};
 
     // ...override specific options for this specific fuzz suite
     mempool_opts.estimator = nullptr;
     mempool_opts.check_ratio = 1;
+    mempool_opts.require_standard = fuzzed_data_provider.ConsumeBool();
```

Comment:
in fa3b8a8d7dddbb6fcf51ac447acc0b1c16bbde20
Not changed by this PR, but why don't we just fuzz with require standard always on? false just disables some tests, no?

Reply:
Is there a reason *not* to fuzz the additional branches?

Reply:
Not strongly opinionated, resolving this comment, but have a question to clarify my understanding. My thinking here was based on the `check_ratio` set to 1, since any other value just means you run `check()` less often. AFAICT it's the same idea with standardness checks; if you turn it off, you're running fewer tests. Is it another branch if the codepaths you can hit with `true` is a superset of the codepaths you can hit with `false`? Am I misunderstanding?

Reply:
I think it is good to also fuzz for non standard transactions, since we support it. Otherwise the validation codepath for non standard transactions that would be executed in "production" would not be fuzzed. On the other hand, "prod" is only testnet here since we don't allow `-acceptnonstdtxn` for mainnet..

nit: Since this is using cycles for a test-only option, maybe we could lower the probability of setting `require_standard` to `false`? Something like `require_standard = (ConsumeIntegralInRange(1, 10) % 10 != 0)`?

Reply:
> AFAICT it's the same idea with standardness checks;

Not really. `check` will not mutate the mempool. However, `require_standard` may leave the mempool in a different state with the same ATMP calls, but a different value of `require_standard`. For example tx-size may create more inputs in the mempool. Or a raw non-standard script may trigger more paths in script evaluation. I think the mempool also uses script caching, so I was hoping for  this to also be a check that the mempool doesn't print coins out of thin air.

Reply:
> check will not mutate the mempool. However, require_standard may leave the mempool in a different state with the same ATMP calls, but a different value of require_standar

Ahh yeah, make sense different branches, thanks!